src/impls/legacy.rs

```diff
@@ -0,0 +1,253 @@
+//! Legacy serialization implementation for G1 and G2 points
+
+use crate::impls::inner_types::*;
+use crate::traits::LegacyG1Point;
+use crate::traits::LegacyG2Point;
+use crate::{BlsError, SerializationFormat};
+
+/// Common bit manipulation constants for legacy serialization
+const INFINITY_BYTE: u8 = 0xc0;
+const MODERN_Y_SIGN_BIT: u8 = 0x20; // Bit 5
+const LEGACY_Y_SIGN_BIT: u8 = 0x80; // Bit 7
+const MODERN_COMPRESSION_BIT: u8 = 0x80; // Bit 7
+const MODERN_FORMAT_MASK: u8 = 0x1f; // Clear top 3 bits
+const LEGACY_FORMAT_MASK: u8 = 0x7f; // Clear bit 7
+const LEGACY_VALIDATION_MASK: u8 = 0xe0; // Check bits 5-7 are clear
+
+/// Convert from modern to legacy format for compressed point serialization
+#[inline]
+fn modern_to_legacy_format(bytes: &mut [u8]) {
+    // Check for infinity point (same in both formats)
+    if bytes[0] == INFINITY_BYTE {
+        return;
+    }
+
+    // Extract y-coordinate sign from modern format (bit 5)
+    let y_sign = (bytes[0] & MODERN_Y_SIGN_BIT) != 0;
+
+    // Clear modern format bits (top 3 bits)
+    bytes[0] &= MODERN_FORMAT_MASK;
+
+    // Set legacy y-coordinate sign (bit 7)
+    if y_sign {
+        bytes[0] |= LEGACY_Y_SIGN_BIT;
+    }
+}
+
+/// Convert from legacy to modern format for compressed point deserialization
+#[inline]
+fn legacy_to_modern_format(bytes: &mut [u8]) -> Result<(), BlsError> {
```

Comment:
@coderabbitai why do we have a stand-alone legacy to modern format function? Does it make more sense for this to be in some other structure?

Reply:
Do you think it makes sense inside SerializationFormat? Or should that not be the case? SerializationFormat is basically just Modern or Legacy. Does it make sense for this logic to be inside that?